Code/Java/RedBlackTree.java

```diff
@@ -0,0 +1,688 @@
+//   Red Black Tree : Insertion and Deletion operations 
+/*A red-black tree is a kind of self-balancing binary search tree where each node has an extra bit, and that bit is often interpreted as the colour (red or black)*/
+// Time complexity of Insertion and deletion operation in Red black tree is O(log n ).
+package rbt;
+import java.util.*;
+class Node {
+	int data;  // contains the data
+	Node parent;  // pointer to the parent
+	Node left;  // pointer to left child
+	Node right; // pointer to right child
+	int color;  // 1-> Red, 0-> Black
+}
+
+
+// class RedBlackTree implements the operations in Red Black Tree
+public class RedBlackTree {
+	private Node root;
+	private Node TNULL;
+
+	private void preOrder(Node node) {
+		if (node != TNULL) {
+			String c;
+			if(node.color == 0)
+			{
+				c = "Black";
+			}
+			else
+			{
+				c ="Red";
+			}
+			System.out.print(node.data+"("+c+")"+ " ");
+			preOrder(node.left);
+			preOrder(node.right);
+		} 
+	}
+
+	private void inOrder(Node node) {
+		if (node != TNULL) {
+			String c;
+			if(node.color == 0)
+			{
```

Comment:
it's a good practice to write the if conditions as:

```java
if(x == 0) {
 .....
}

```

rather than 

```java
if (x == 0)
{

}
```

so do update this in the code